python/pyspark/sql/utils.py

```diff
@@ -61,6 +61,8 @@ def deco(*a, **kw):
                                              e.java_exception.getStackTrace()))
             if s.startswith('org.apache.spark.sql.AnalysisException: '):
                 raise AnalysisException(s.split(': ', 1)[1], stackTrace)
+            if s.startswith('org.apache.spark.sql.catalyst.analysis.NoSuchTableException: '):
```

Comment:
I think we should either add all or nothing. I kind of prefer nothing actually because it's less code to maintain.